examples/waspleau/Notes.md

```diff
@@ -0,0 +1,198 @@
+
+I was attempting to return hard coded JSON from a query without any entities defined, but ran into the following issue on `wasp start`:
+
+```
+Server: /Users/shayne/dev/wasp/examples/waspleau/.wasp/out/server/node_modules/.prisma/client/index.js:3
+Server (stderr):     throw new Error(
+Server (stderr):           ^
+Server (stderr):
+Server (stderr): Error: @prisma/client did not initialize yet. Please run "prisma generate" and try to import it again.
+Server (stderr): In case this error is unexpected for you, please report it in https://github.com/prisma/prisma/issues
+Server (stderr):     at new PrismaClient (/Users/shayne/dev/wasp/examples/waspleau/.wasp/out/server/node_modules/.prisma/client/index.js:3:11)
+Server (stderr):     at createDbClient (file:///Users/shayne/dev/wasp/examples/waspleau/.wasp/out/server/src/dbClient.js:5:24)
+Server (stderr):     at file:///Users/shayne/dev/wasp/examples/waspleau/.wasp/out/server/src/dbClient.js:11:18
+Server (stderr):     at ModuleJob.run (internal/modules/esm/module_job.js:152:23)
+Server (stderr):     at async Loader.import (internal/modules/esm/loader.js:166:24)
+Server (stderr):     at async Object.loadESM (internal/process/esm_loader.js:68:5)
+Server (stderr): [nodemon] app crashed - waiting for file changes before starting...
+```
+
+Therefore, I assumed I needed to migrate, however...
+
+### Running `wasp db migrate-dev` without any entities defined
+
+```
+Database successfully set up!
+Copying migrations folder from Wasp to Prisma project...
+Done copying migrations folder.
+Performing migration...
+Prisma schema loaded from ../db/schema.prisma
+Datasource "db": SQLite database "dev.db" at "file:./dev.db"
+
+Already in sync, no schema change or pending migration was found.
+
+Running generate... (Use --skip-generate to skip the generators)
+Error:
+You don't have any models defined in your schema.prisma, so nothing will be generated.
+You can define a model like this:
+
+model User {
+  id    Int     @id @default(autoincrement())
+  email String  @unique
+  name  String?
+}
+
+More information in our documentation:
+https://pris.ly/d/prisma-schema
+
+Error: Migrate dev failed: Migrate (dev) failed with exit code: 1
+```
+
+The fix was to add a dummy migration just to get past the Prisma error.
+
+
+| Bug captured here: https://github.com/wasp-lang/wasp/issues/398 |
+| --------------------------------------------------------------- |
+
+## 2) Axios version pinning
+
+```
+wasp: Error: Dependency conflict for user npm dependency (axios, ^0.21.1): Version must be set to the exactly the same version as the one wasp is using: ^0.21.1
+CallStack (from HasCallStack):
+  error, called at src/Generator/WebAppGenerator.hs:73:31 in waspc-0.2.2.2-B3XOTY4be7eFFZyFqJsQfH:Generator.WebAppGenerator
+```
+
+## 3) Dockerfile keeps getting overwritten. Any way to add to it? Do we want to use Docker Compose to allow for additional services like Redis?
```

Comment:
Ha good question -> for now there is no way to add to it. Ideally we would have support for this in Wasp, if needed. What was your use case? You wanted to spin up a Redis as a separate micro service? Or just install it on the machine that server runs on?

Reply:
I think there were two things to consider here, one being using docker compose for additional services like Redis or DB or nginx in front. The other is if for some reason the Docker file out of the box isn't quite working for the user, or they wanted to tweak it in some way. Not a must have, just something I noticed and to consider at some point in future.

Reply:
Makes sense, I think it is certainly something we have to consider in the future and figure out how to support it.

Should we create a separate issue for this? Or do we leave it here, assuming we will at some point revisit these notes and then create appropriate issues? I am not sure myself I have to admit.